designer/server/src/main/scala/pl/touk/nussknacker/ui/api/description/DeploymentApiEndpoints.scala

```diff
@@ -244,10 +253,12 @@ object DeploymentApiEndpoints {
         case DeployValidationError(message)       => message
       }
 
-    implicit val conflictingDeploymentIdErrorCodec: Codec[String, ConflictingDeploymentIdError, CodecFormat.TextPlain] =
-      BaseEndpointDefinitions.toTextPlainCodecSerializationOnly[ConflictingDeploymentIdError](err =>
-        s"Deployment with id ${err.id} already exists"
-      )
+    implicit val conflictingDeploymentIdErrorCodec: Codec[String, ConflictRunDeploymentError, CodecFormat.TextPlain] =
+      BaseEndpointDefinitions.toTextPlainCodecSerializationOnly[ConflictRunDeploymentError] {
+        case ConflictingDeploymentIdError(id) => s"Deployment with id $id already exists"
+        case ConcurrentDeploymentsForScenarioArePerformedError(concurrentDeploymentsIds) =>
+          s"Deployment can't be run because deployment with ids ${concurrentDeploymentsIds.toList.mkString(", ")} for the same scenario are performed concurrently"
```

Comment:
Yeah I had the same consternation but in some specific situations (e.g. some bug, some migrations) it could be > 1 deployment. The proposed comment fits better. Changed.